seva-web/lib/settings_page.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'websocket.dart';
+import 'dart:async';
+import 'dart:html';
+import 'dart:convert';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+class WebProxy extends StatefulWidget {
+  @override
+  State<WebProxy> createState() => _WebProxyState();
+}
+
+class _WebProxyState extends State<WebProxy> {
+  var http_textarea = TextEditingController();
+  var no_proxy_textarea = TextEditingController();
+
+  final _form = GlobalKey<FormState>();
+
+  bool waiting_on_response_ = false;
+
+  Future<WebSocketCommand> response_handler() async {
+    // catch the response code and update state accordingly
+    setState(() {
+      waiting_on_response_ = true;
+    });
+    String response = await stream.first;
+    setState(() {
+      waiting_on_response_ = false;
+    });
+    return WebSocketCommand.from_json(jsonDecode(response));
+  }
+
+  Future<void> write_proxy(var serialized_settings) async {
+    // writes proxy
+    print(serialized_settings);
+    WebSocketCommand.outbound("save_settings", [serialized_settings]).send();
+    WebSocketCommand command = await response_handler();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: const Text("Web-Proxy Settings"),
+          backgroundColor: Colors.redAccent,
+        ),
+        body: Form(
+          key: _form, //assigning key to form
+          child: Column(
+            children: [
+              TextFormField(
+                controller: http_textarea,
+                keyboardType: TextInputType.multiline,
+                maxLines: 5,
+                decoration: const InputDecoration(
+                    hintText: "Enter your Http Proxy Settings",
+                    focusedBorder: OutlineInputBorder(
+                        borderSide:
+                            BorderSide(width: 1, color: Colors.redAccent))),
+                validator: (value) {
```

Comment:
Bad validator, empty response is acceptable, but if there is a response it has to be a valid URL. Look into URL validation tools in flutter.